dbt/scripts/utils/export.py

```diff
@@ -40,6 +48,69 @@ def export_models(
         * rebuild (bool): Rebuild models before exporting, defaults to False
         * where (str): Optional SQL expression representing a WHERE clause to
             filter models
+        * output_dir (str): Optional Unix path to directory where output files
+            should be stored
+
+    Returns a list of pathlib.Path objects representing the paths to the files
+    that the function created.
+    """
+    models_for_export = query_models_for_export(
+        select=select,
+        selector=selector,
+        rebuild=rebuild,
+        where=where,
+    )
+    output_paths: list[pathlib.Path] = []
+    for model in models_for_export:
+        output_path = save_model_to_workbook(model, output_dir)
+        output_paths.append(output_path)
+    return output_paths
```

Comment:
I don't feel 100% great about the way that I've cut up this function to expose internals that `export_qc_town_close_reports` can use to modify the query data. In particular, I think it's kind of cumbersome to deal with the `ModelForExport` abstraction that acts as the interface between the `query` and `save` functions. I'm curious if you have thoughts on a better way to separate the concerns.

Reply:
Kind of a bizarre idea, but maybe something like:

- Have a `query` for each model
- Have a `filter` for each model

Create a unique set of all the `query` values across models, run them, then populate the models using the results (and apply the filter to each one).

Might be more complicated? Maybe save it for the next incremental improvement.

Reply:
I like this idea! I'll open up an issue witih this design as an iterative improvement, and then pick it up once we've validated that the ability to run reports for all towns is an important feature for the project.

Reply:
Issue open here: https://github.com/ccao-data/data-architecture/issues/637